src/models/image-map.ts

```diff
@@ -204,38 +250,70 @@ export const ImageMapModel = types
       addFileImage: flow(function* (file: File): Generator<PromiseLike<any>, Promise<ImageMapEntryType>, unknown> {
         const simpleImage = (yield storeFileImage(_db, file)) as ISimpleImage;
         const { normalized } = parseFauxFirebaseRTDBUrl(simpleImage.imageUrl);
-        const entry: ImageMapEntrySnapshot = {
+        const entry: IImageHandlerStoreResult = {
                 filename: file.name,
                 contentUrl: normalized,
-                displayUrl: simpleImage.imageData
+                displayUrl: simpleImage.imageData,
+                success: true
               };
-        return self.addImage(entry.contentUrl!, entry);
+        return self._addImage(entry.contentUrl!, entry);
       }),
 
-      getImage: flow(function* (url: string, options?: IImageBaseOptions)
+      _storeAndAddImage: flow(function* (url: string, handler: IImageHandler, options?: IImageBaseOptions)
                        : Generator<PromiseLike<any>, ImageMapEntryType | Promise<ImageMapEntryType>, unknown> {
-        if (!url) {
-          return clone(self.images.get(placeholderImage)!);
-        }
+        const storeResult = (yield handler.store(url, { db: _db, ...options })) as IImageHandlerStoreResult;
+        return self._addImage(url, storeResult);
+      }),
+    };
+  })
+  .actions(self => ({
+    // eslint-disable-next-line require-yield
+    getImage: flow(function* (url: string, options?: IImageBaseOptions)
+                    : Generator<PromiseLike<any>, ImageMapEntryType | Promise<ImageMapEntryType>, unknown> {
+      if (!url) {
+        // TODO: how often does this happen, should it be silently ignored like this?
+        return clone(self.images.get(placeholderImage)!);
+      }
 
-        const imageEntry = self.images.get(url);
-        if (imageEntry) {
-          // This might or might not have dimensions yet
-          return imageEntry;
-        }
+      const imageEntry = self.images.get(url);
+      if (imageEntry?.status === EntryStatus.Ready) {
+        // This image has been downloaded successfully already
+        return imageEntry;
+      }
 
-        const handler = self.getHandler(url);
-        if (handler) {
-          const imageEntrySnapshot = (yield handler.store(url, { db: _db, ...options })) as ImageMapEntrySnapshot;
-          return self.addImage(url, imageEntrySnapshot);
-        }
-        else {
-          return clone(self.images.get(placeholderImage)!);
-        }
-      })
+      const existingStoringPromise = self.storingPromises[url];      
+      if (existingStoringPromise && imageEntry?.status !== EntryStatus.Error) {
+        // If the imageEntry is errored we ignore the existing promise
+        // This way a second getImage request will try to store the image again
```

Comment:
Yes that is a good point. I haven't tested the error path to see how it interacts with the MobX observer approach used by the drawing tile. Now that I'm thinking about it, it seems possible it could cause an infinite loop. 

In the case of the promise based stuff the other tiles are doing, I think we should be safe because I think they only call getImage one time.

I'll add a TODO here in the code, and I added TODO on the drawing tile PR.